### ISSUE TITLE
Do not abort sync on unavailable federated shares

### ISSUE THREAD

**Issue by ckamm** [owncloud/client#7375]
From #5088:

If there's a federated share but the source server is unavailable, the server will reply 503 with something like this:

```
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\ServiceUnavailable</s:exception>
  <s:message>Storage is temporarily not available</s:message>
</d:error>
```

Currently the client assumes 503 with Sabre\DAV\Exception\ServiceUnavailable means maintenance mode. That's not a correct assumption for this case.

It can't rely on the "s:message" string, since it's translated. The client should request status.php to check for maintenance mode before continuing.

Test criteria: Unavailable federated shares don't cause the sync run to abort. Alll other files should be synced normally, only the share should be marked with an error.

**Comment by michaelstingl** [comment]
@ckamm Needs also to be fixed on the ownCloud server or in Sabre\DAV?

/cc @DeepDiver1975 @PVince81 @micbar

**Comment by ckamm** [comment]
No, I don't think so. The client using this response to detect maintenance mode was a workaround/hack. It should check the existing API for it.
